db/upgrade.php

```diff
@@ -65,5 +67,54 @@ function xmldb_tool_mergeusers_upgrade ($oldversion) {
         upgrade_plugin_savepoint(true, 2013112912, 'tool', 'mergeusers');
     }
 
+    if ($oldversion < 2023040401) {
+
+        // Define field mergedbyuserid to be added to tool_mergeusers.
+        $table = new xmldb_table('tool_mergeusers');
+        $field = new xmldb_field('mergedbyuserid', XMLDB_TYPE_INTEGER, '10', null, null, null, null, 'success');
+
+        // Conditionally launch add field mergedbyuserid.
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        // Mergeusers savepoint reached.
+        upgrade_plugin_savepoint(true, 2023040401, 'tool', 'mergeusers');
+    }
+    
+    if ($oldversion < 2023062310) {
+        // Define table tool_mergeusers_queue to be created.
+        $table = new xmldb_table('tool_mergeusers_queue');
+        // Adding fields to table tool_mergeusers_queue.
+        $table->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
+        $table->add_field('removeuserfield', XMLDB_TYPE_CHAR, '50', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('removeuservalue', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('removeuserid', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+        $table->add_field('keepuserfield', XMLDB_TYPE_CHAR, '50', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('keepuservalue', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('keepuserid', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+        $table->add_field('mergedbyuserid', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+        $table->add_field('timeadded', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('timemodified', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+        $table->add_field('status', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('retries', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('log', XMLDB_TYPE_TEXT, null, null, null, null, null);
+        // Adding keys to table tool_mergeusers_queue.
+        $table->add_key('primary', XMLDB_KEY_PRIMARY, ['id']);
+        $table->add_key('removeuserid', XMLDB_KEY_FOREIGN, ['removeuserid'], 'user', ['id']);
```

Comment:
with this foreign key an index has created with the following name
mdl_toolmergqueu_rem2_ix